gulpfile.js

```diff
@@ -0,0 +1,46 @@
+'use strict';
+
+var gulp = require('gulp');
+var sass = require('gulp-sass');
+var webpack = require('gulp-webpack');
+var uglify = require('gulp-uglify');
+var minifyCSS = require('gulp-minify-css');
+var minifyHTML = require('gulp-minify-html');
+
+gulp.task('sass', function() {
+  gulp.src('./app/sass/**/*.scss')
+    .pipe(sass.sync().on('error', sass.logError))
+    .pipe(minifyCSS({compaibility: 'ie8'}))
+    .pipe(gulp.dest('./public/css'));
+});
+
+gulp.task('sass:watch', function() {
+  gulp.watch('./app/sass/**/*.scss', ['sass']);
+});
+
+gulp.task('webpack', function() {
+  return gulp.src('app/js/**/*.js')
+    .pipe(webpack({
+      output: {
+        filename: 'super.js'
+      }
+     }))
+      .pipe(uglify())
+      .pipe(gulp.dest('public/js/'));
+});
+
+gulp.task('copy', function() {
+  var opts = {
+    conditionals: true,
+    spare: true
+  };
+
+  return gulp.src('app/**/*.html')
+    .pipe(gulp.dest('public/'))
+    .pipe(minifyHTML(opts))
```

Comment:
gulp needs to be explicit, right now the first destination is different from the second destination, and actually not going anywhere. You need the `./` in front of the directory name.